xbmc/pvr/windows/GUIWindowPVRBase.cpp

```diff
@@ -838,7 +833,8 @@ void CGUIWindowPVRBase::UpdateButtons(void)
 
 void CGUIWindowPVRBase::UpdateSelectedItemPath()
 {
-  m_selectedItemPaths.at(m_bRadio) = m_viewControl.GetSelectedItemPath();
+  if (!m_viewControl.GetSelectedItemPath().empty())
```

Comment:
`UpdateSelectedItemPath()` is called several times without a valid `m_viewControl.GetSelectedItemPath()`. Thats the reason why sometimes the wrong channel (the first item) is selected.